Calibration/PCCAlCaRecoProducers/plugins/AlcaPCCProducer.cc

```diff
@@ -0,0 +1,115 @@
+{
+    resetNLumi_ = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getUntrackedParameter<int>("resetEveryNLumi",-1);
+    fPixelClusterLabel = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getParameter<edm::InputTag>("pixelClusterLabel");
+    trigstring_ = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getUntrackedParameter<std::string>("trigstring","alcaPCC");
+
+    ftotalevents = 0;
+    countLumi_ = 0;
+    beginLumiOfPCC_ = endLumiOfPCC_ = -1;
+
+    produces<reco::PixelClusterCounts, edm::InLumi>(trigstring_);
+    pixelToken=consumes<edmNew::DetSetVector<SiPixelCluster> >(fPixelClusterLabel);
+}
+
+//--------------------------------------------------------------------------------------------------
+AlcaPCCProducer::~AlcaPCCProducer(){
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
+    ftotalevents++;
+
+    unsigned int bx=iEvent.bunchCrossing();
+    //std::cout<<"The Bunch Crossing"<<bx<<std::endl;
+    thePCCob->eventCounter(bx);
+
+    //Looping over the clusters and adding the counts up  
+    edm::Handle< edmNew::DetSetVector<SiPixelCluster> > hClusterColl;
+    iEvent.getByToken(pixelToken,hClusterColl);
+
+    const edmNew::DetSetVector<SiPixelCluster>& clustColl = *(hClusterColl.product()); 
+    // ----------------------------------------------------------------------
+    // -- Clusters without tracks
+    for (auto const & mod: clustColl) {
+        if(mod.empty()) { continue; }
+        DetId detId = mod.id();
+
+        // -- clusters on this det
+        edmNew::DetSet<SiPixelCluster>::const_iterator  di;
+        int nClusterCount=0;
+        for (di = mod.begin(); di != mod.end(); ++di) {
+            nClusterCount++;
+        }
+        int nCluster = mod.size();
+        if(nCluster!=nClusterCount) {
+            std::cout<<"counting yields "<<nClusterCount<<" but the size is "<<nCluster<<"; they should match."<<std::endl;
+        }
+        thePCCob->increment(detId(), bx, nCluster);
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
+    //New PCC object at the beginning of each lumi section
+    thePCCob = std::make_unique<reco::PixelClusterCounts>();
+    
+    if ( countLumi_ == 0 || (resetNLumi_ > 0 && countLumi_%resetNLumi_ == 0) ) {
+        beginLumiOfPCC_ = lumiSeg.luminosityBlock();
+    }
+
+    countLumi_++;
+
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
+    //Saving the PCC object 
+    //FIXME! The line below doesn't work but I need to save here. 
```

Comment:
I would like to save the object here because as of now the increment method is lost when resetNLumi_ isn't reached at the end of the set of lumi sections.  Example: I have 20 LS and I define my block at 7. So far I save 1-14 and 15-20 isn't saved.